cypress/e2e/permit2.test.ts

```diff
@@ -63,144 +52,140 @@ describe('Permit2', () => {
         approval.setTokenAllowanceForPermit2({ owner: wallet, token: INPUT_TOKEN })
         approval.setPermit2Allowance({ owner: wallet, token: INPUT_TOKEN })
       })
-      .then(swaps)
+      .then(initiateSwap)
+      .then(() => {
+        cy.get(getTestSelector('confirmation-close-icon')).click()
+        // Verifies that there is a successful swap notification.
+        cy.contains('Swapped').should('exist')
+      })
   })
 
   it('swaps after completing full permit2 approval process', () => {
-    cy.get(APPROVE_BUTTON)
-      .click()
-      .then(() => {
-        const approvalTime = Date.now()
-        cy.get(APPROVE_BUTTON).should('have.text', 'Approval pending')
+    initiateSwap()
+    cy.get(getTestSelector('PendingModalContent-title')).should('have.text', 'Approve permit')
+    cy.contains('Approved').should('exist')
 
-        // There should be a successful Approved notification.
-        cy.contains('Approved').should('exist')
+    cy.get(getTestSelector('PendingModalContent-title')).should('have.text', 'Approve DAI')
+    cy.get(getTestSelector('PendingModalContent-title')).should('have.text', 'Confirm Swap')
 
-        swaps()
+    const approvalTime = Date.now()
 
-        expectTokenAllowanceForPermit2ToBeMax()
-        expectPermit2AllowanceForUniversalRouterToBeMax(approvalTime)
-      })
+    cy.contains('Swapped').should('exist')
+
+    expectTokenAllowanceForPermit2ToBeMax()
+    expectPermit2AllowanceForUniversalRouterToBeMax(approvalTime)
   })
 
-  it('swaps after handling user rejection of approvals and signatures', () => {
+  it('swaps after handling user rejection of both approval and signature', () => {
     const USER_REJECTION = { code: 4001 }
     cy.hardhat().then((hardhat) => {
       const tokenApprovalStub = cy.stub(hardhat.wallet, 'sendTransaction')
       tokenApprovalStub.rejects(USER_REJECTION) // reject token approval
-
       const permitApprovalStub = cy.stub(hardhat.provider, 'send')
       permitApprovalStub.withArgs('eth_signTypedData_v4').rejects(USER_REJECTION) // reject permit approval
       permitApprovalStub.callThrough() // allows non-eth_signTypedData_v4 send calls to return non-stubbed values
 
-      // Clicking the approve button should trigger a token approval that will be rejected by the user (tokenApprovalStub).
-      cy.get(APPROVE_BUTTON).click()
+      initiateSwap()
 
-      // The swap component should prompt approval again.
-      cy.get(APPROVE_BUTTON)
-        .should('have.text', `Approve use of ${INPUT_TOKEN.symbol}`)
+      // tokenApprovalStub should reject here, and the modal should revert to the review state.
+      cy.contains('Review Swap')
+        .should('be.visible')
         .then(() => {
```

Comment:
This shouldn't be necessary.

```suggestion
cy.then(() => {
```

Reply:
removed this chaining